File: src/main/java/com/rimlang/rim/syntax/WhileSyntaxNode.java
package com.rimlang.rim.syntax;

import java.util.List;

public class WhileSyntaxNode extends SyntaxNode {
    private final BooleanExpression booleanExpression;
    private final List<SyntaxNode> body;

    public WhileSyntaxNode(BooleanExpression booleanExpression, List<SyntaxNode> body) {
        this.booleanExpression = booleanExpression;
        this.body = body;
    }
}

File: src/main/kotlin/com/rimlang/rim/syntax/node/control/WhileSyntaxNode.kt
package com.rimlang.rim.syntax.node.control

import com.rimlang.rim.lexer.token.Token
import com.rimlang.rim.syntax.node.expression.BooleanExpression
import com.rimlang.rim.syntax.node.syntax.SyntaxNode

data class WhileSyntaxNode(
    val booleanExpression: BooleanExpression,
    val body: List<SyntaxNode>,
    override val firstToken: Token
) : SyntaxNode(firstToken)
